inmaputil/download.go

```diff
@@ -0,0 +1,105 @@
+package inmaputil
+
+import (
+	"github.com/mholt/archiver"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/cavaliercoder/grab"
+)
+
+func Download(path string) string {
+	// Check if local file exists
+	if _, err := os.Stat(path); !os.IsNotExist(err) {
+		return path;
+	} 
+
+	// Check if path doesn't start with http or https
+	if !strings.HasPrefix(path, "http://") && !strings.HasPrefix(path, "https://") {
+		return path;
+	}
+
+	// Prepare temporary directory
+	if err := os.MkdirAll("./temp/", os.ModePerm); err != nil {
+		fmt.Fprintf(os.Stderr, "Failed creating temporary folder to download file. %v\n", err)
+		return path;
+	}
+
+	// Download
+	client := grab.NewClient()
+	req, _ := grab.NewRequest("./temp/", path)
+
+	fmt.Printf("Downloading %v...\n", req.URL())
```

Comment:
Instead of writing log messages directly to the console, instead send them to a channel that's supplied by the caller, so the caller can decide what to do with them.

You can see an example implementation [here](https://github.com/spatialmodel/inmap/blob/master/run.go#L153), [here](https://github.com/spatialmodel/inmap/blob/master/run.go#L213), and [here](https://github.com/spatialmodel/inmap/blob/master/inmaputil/inmap.go#L136)